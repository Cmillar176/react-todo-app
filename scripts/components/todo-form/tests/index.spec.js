import React from "react";
import { shallow } from "enzyme";
import TodoForm from "../";

const renderComponent = function(props) {
  return shallow(<TodoForm {...props} />);
};

describe("TodoForm", () => {
  let componentWrapper;
  let addTodoSpy;
  let setUserInputSpy;

  beforeEach(() => {
    addTodoSpy = jest.fn();
    setUserInputSpy = jest.fn();

    componentWrapper = renderComponent({
      addTodo: addTodoSpy,
      setUserInput: setUserInputSpy,
      userInput: "Pick up milk"
    });
  });

  it("renders correctly", () => {
    expect(componentWrapper).toMatchSnapshot();
  });

  describe("when text is added to the input field", () => {
    let inputWrapper;

    beforeEach(() => {
      inputWrapper = componentWrapper.find("#todoText");
      inputWrapper.simulate("change", {
        target: {
          value: "Get bread"
        }
      });
    });

    it("triggers relevant callback with correct value", () => {
      expect(setUserInputSpy).toHaveBeenCalledTimes(1);
      expect(setUserInputSpy).toBeCalledWith("Get bread");
    });
  });

  describe("when the form is submitted", () => {
    let formWrapper;

    describe("when the user input is blank", () => {
      beforeEach(() => {
        componentWrapper.setProps({ userInput: "" });
        formWrapper = componentWrapper.find("form");
        formWrapper.simulate("submit", { preventDefault: jest.fn() });
      });

      it("does not trigger relevant callback", () => {
        expect(addTodoSpy).toHaveBeenCalledTimes(0);
      });
    });

    describe("when the user input is populated", () => {
      beforeEach(() => {
        componentWrapper.setProps({ userInput: "Buy vitamins" });
        formWrapper = componentWrapper.find("form");
        formWrapper.simulate("submit", { preventDefault: jest.fn() });
      });

      it("triggers relevant callback with correct value", () => {
        expect(addTodoSpy).toHaveBeenCalledTimes(1);
        expect(addTodoSpy).toBeCalledWith("Buy vitamins");
      });
    });
  });
});
