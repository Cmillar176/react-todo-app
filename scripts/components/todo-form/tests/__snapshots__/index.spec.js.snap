// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders correctly 1`] = `
<div
  className="todo-form__wrapper"
>
  <h1>
    Add a Todo
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="todoText"
      >
        Text
      </label>
      <input
        className="form-control"
        id="todoText"
        onChange={[Function]}
        placeholder="Enter text here"
        type="text"
        value="Pick up milk"
      />
    </div>
    <button
      className="btn btn-default"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
